Generate changelog for 3.3.3 (#11749)
tarantool/tarantool: changelogs/3.3.3.md

# 3.3.3

Date: 2025-08-12
Tag: 3.3.3

## Overview

Tarantool 3.x is the recommended release series. Users of Tarantool 2.11 are
encouraged to update to the latest 3.x release.

This release resolves 29 bugs and introduces 1 tooling improvement since 3.3.2.

> Please, consider the full list of user-visible changes below.

Reviewer: ```suggestion
Please consider the full list of user-visible changes below.
```

## Compatibility

Tarantool 2.x and 3.x are compatible in the binary data layout, client-server
protocol, and replication protocol. It means that the updating may be performed

> with zero downtime for read requests and the-order-of-network-lag downtime for

Reviewer: ```suggestion
with zero downtime for read requests and the order-of-network-lag downtime for
```

write requests.

Please, follow the [upgrade procedure][upgrade] to plan your update actions.

Users of Tarantool 2.x may be interested in the [compat][compat] options that
allow to imitate some 2.x behavior. This allows to perform application code
update step-by-step, not all-at-once.

[compat]: https://www.tarantool.io/en/doc/latest/reference/configuration/configuration_reference/#compat

[upgrade]: https://www.tarantool.io/en/doc/latest/book/admin/upgrades/

## Functionality added or changed

### Tools

* It is now possible to call LuaJIT's platform profile function

>   `misc.sysprof.report()` during the profiling as well (gh-11229).

Reviewer 2: We can coalesce it with the Tools section below (in any way).

## Bugs fixed

### Core

* Fixed a bug where passwords are leaked into logs
  when using `listen` and `bootstrap_leader` options.
* Now the server binds all URIs matching the `listen` parameter in the box

>   configuration. Previously it bound only a single URI per entry. Now providing

Reviewer: ```suggestion
  configuration. Previously, it bound only a single URI per entry. Now providing
```

>   a single port number makes all interfaces be listened on that port. It is

Reviewer: ```suggestion
  a single port number makes all interfaces listen on that port. It is
```
